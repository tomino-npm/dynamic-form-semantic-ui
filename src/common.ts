import { FormElement, DataSet } from '@tomino/dynamic-form';
import { initCss } from '@tomino/toolbelt';

export type SignatureType = {
  comment: string;
  signature: string;
  verifiedState: 'Pending' | 'Verified' | 'Rejected';
  rejected?: boolean;
  uid: string;
  name: string;
  date: Date;
  setValue?(name: string, value: any): any;
};

export type SignatureHandlers = {
  validateForm(): boolean;
  verifySignature(uid: string, signature: string): Promise<string>;
  sign(
    source: string,
    reject: boolean,
    password: string,
    reason: string,
    submit: boolean
  ): Promise<SignatureType>;
  signatureFont(): string;
};

export type FormControlProps = {
  formControl: FormElement;
  owner: DataSet;
  handlers?: { [index: string]: any };
  readOnly: boolean;
};

export const { css, renderCss } = initCss('form-');
