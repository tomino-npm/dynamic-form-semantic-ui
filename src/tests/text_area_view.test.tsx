import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { create } from './form_query_data';
import { JSONSchema, FormDefinition, FormModel } from '@tomino/dynamic-form';
import { TestComponent } from './common';

describe('Form', () => {
  const schema: JSONSchema = {
    type: 'object',
    properties: {
      description: {
        type: 'string'
      },
      empty: {
        type: 'string'
      }
    },
    required: ['name']
  };

  const formDefinition: FormDefinition = create.form({
    elements: [
      create.formElement({
        row: 0,
        column: 0,
        width: 16,
        control: 'Textarea',
        label: 'Description',
        source: 'description'
      }),
      create.formElement({
        row: 1,
        column: 0,
        width: 16,
        control: 'Textarea',
        label: 'Description',
        source: 'empty',
        controlProps: {
          placeholder: 'Render me!'
        }
      })
    ]
  });

  const formData = {
    description: 'Some Description'
  };

  describe('TextArea', () => {
    function componentWithData() {
      const form = new FormModel(formDefinition, schema, formData);

      // just another notation
      return <TestComponent form={form} />;
    }

    it('renders correctly', () => {
      const component = renderer.create(componentWithData());
      expect(component).toMatchSnapshot();
    });

    return { componentWithData };
  });
});
