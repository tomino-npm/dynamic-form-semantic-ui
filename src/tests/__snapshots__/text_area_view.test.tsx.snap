// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea renders correctly 1`] = `
<div
  className="ui segment ui form"
>
  <div
    className="ui dividing header"
  >
    Standard
  </div>
  <style>
    
  </style>
  <div
    className="ui form "
  >
    <div
      className="fields"
    >
      <div
        className="sixteen wide field"
      >
        <label
          htmlFor="description"
        >
          Description
        </label>
        <textarea
          error=""
          name="description"
          onChange={[Function]}
          onInput={[Function]}
          placeholder={null}
          rows={3}
          style={
            Object {
              "resize": "",
            }
          }
          value="Some Description"
        />
      </div>
    </div>
    <div
      className="fields"
    >
      <div
        className="sixteen wide field"
      >
        <label
          htmlFor="empty"
        >
          Description
        </label>
        <textarea
          error=""
          name="empty"
          onChange={[Function]}
          onInput={[Function]}
          placeholder="Render me!"
          rows={3}
          style={
            Object {
              "resize": "",
            }
          }
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="ui dividing header"
  >
    Readonly
  </div>
  <style>
    
  </style>
  <div
    className="ui form form-hjcaghiia"
  >
    <div
      className="fields"
    >
      <div
        className="sixteen wide field"
      >
        <label
          htmlFor="description"
        >
          Description
        </label>
        <textarea
          disabled={true}
          error=""
          name="description"
          onChange={[Function]}
          onInput={[Function]}
          placeholder={null}
          rows={3}
          style={
            Object {
              "resize": "",
            }
          }
          value="Some Description"
        />
      </div>
    </div>
    <div
      className="fields"
    >
      <div
        className="sixteen wide field"
      >
        <label
          htmlFor="empty"
        >
          Description
        </label>
        <textarea
          disabled={true}
          error=""
          name="empty"
          onChange={[Function]}
          onInput={[Function]}
          placeholder="Render me!"
          rows={3}
          style={
            Object {
              "resize": "",
            }
          }
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
